### PR TITLE
Implement dark theme

### PR DESCRIPTION
**Pull request by sofia-chuieva**
- by clicking on switch, theme changes
- "darkMode" and "isChecked" items in sessionStorage were made in order to persist state of Switch component and the theme itself when page refreshes
- implemented Styled Badge for avatar

TRELLO:  https://trello.com/c/ckXk91fg/5-implement-dark-light-mode-switch

https://user-images.githubusercontent.com/80347696/197759613-2ea9b927-faaa-44bb-8201-2fa42a096c11.mp4

### CODE REVIEW

File: js/chat_app/src/components/BadgeAvatar.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
```
Comment (zoolamyke) [inline js/chat_app/src/components/BadgeAvatar.js:1]: Please, clean-up the PR from redundant files.

File: js/chat_app/src/App.js [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
 import { ProtectedRoute } from "./components/ProtectedRoute";
 
-function App() {
+export const App = () => {
+  const { setDarkMode, setChecked } = useContext(ThemeContext);
   const session = JSON.parse(sessionStorage.getItem("user-info"));
   const [auth, setAuth] = useState(session);
   const navigate = useNavigate();
 
+  useEffect(() => {
```
Comment (zoolamyke) [inline js/chat_app/src/App.js:15]:
```
themeStorage.ts

export const get = () => {
  return JSON.parse(sessionStorage...)
}

export const set = (theme: 'dark' | 'light') => {
  sessionStorage.setItem(JSON.strg)
}
```

File: js/chat_app/src/components/DashBoard.js [hunk 3]
```diff
@@ -34,13 +42,17 @@ const DashBoard = () => {
             >
               {`Welcome, ${auth?.user.username}`}
             </Typography>
-            <Switch color="default" />
+            <Switch
+              color="default"
+              onChange={toggleMode}
+              checked={checked}
```
Comment (zoolamyke) [inline js/chat_app/src/components/DashBoard.js:48]:
```suggestion
              checked={darkMode}
```

File: js/chat_app/src/components/DashBoard.js [hunk 4]
```diff
@@ -3,10 +3,16 @@ import { AppBar, Avatar, Toolbar, Typography, Switch } from "@mui/material";
...
+import { ThemeContext } from "./ThemeContext";
+import { LogoutDialog } from "./LogoutDialog";
+import { BadgeAvatar } from "./BadgeAvatar";
+import Brightness4RoundedIcon from "@mui/icons-material/Brightness4Rounded";
+import Brightness5RoundedIcon from "@mui/icons-material/Brightness5Rounded";
 
-const DashBoard = () => {
+export const DashBoard = () => {
   const { auth } = useContext(UserContext);
+  const { toggleMode, checked } = useContext(ThemeContext);
```
Comment (zoolamyke) [inline js/chat_app/src/components/DashBoard.js:14]:
```suggestion
  const { toggleMode, isDarkMode } = useContext(ThemeContext);
```

File: js/chat_app/src/components/ThemeContext.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    },
+  });
+
+  const toggleMode = (e) => {
+    setChecked(e.target.checked);
+    sessionStorage.setItem("isChecked", !checked);
+    setDarkMode(!darkMode);
+    sessionStorage.setItem("darkMode", !darkMode);
+  };
+
```
Comment (zoolamyke) [inline js/chat_app/src/components/ThemeContext.js:28]:
```
React.useEffect(() => {
   sessionStorage.setItem("darkMode", darkMode);
}, [darkMode])
```

File: js/chat_app/src/components/themeStorage.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export const getStorage = (name: string) => {
```
Comment (miromyke) [inline js/chat_app/src/components/themeStorage.ts]: `getThemeFromStorage` or (better) `getTheme`, or (abstract) `get`, or `getStoredTheme`:
```
import {get as getTheme} from 'themeStorage'
```

Comment (miromyke) [reply js/chat_app/src/components/themeStorage.ts]: Same issue with `setStorage` - naming is not great.

File: js/chat_app/package.json [hunk 7]
```diff
@@ -18,6 +20,7 @@
     "react-router-dom": "^6.4.2",
     "react-scripts": "5.0.1",
     "react-use": "^17.4.0",
+    "typescript": "^4.8.4",
```
Comment (miromyke) [inline js/chat_app/package.json:23]: What is the purpose of `typescript` dependency?

**Pull request merged**